cupy/core/core.pyx

```diff
@@ -1098,10 +1098,18 @@ cdef class ndarray:
         cdef Py_ssize_t i, j, offset, ndim, n_newaxes, n_ellipses, ellipsis
         cdef Py_ssize_t ellipsis_sizem, s_start, s_stop, s_step, dim, ind
         cdef vector.vector[Py_ssize_t] shape, strides
-        if not isinstance(slices, tuple):
-            slices = [slices]
-        else:
+        if isinstance(slices, tuple):
             slices = list(slices)
+        elif isinstance(slices, list):
+            slices = list(slices)  # copy list
+            single_integer_array_indexing = True
+            for s in slices:
+                if not isinstance(s, int):
+                    single_integer_array_indexing = False
+            if single_integer_array_indexing:
+                slices = [slices]
```

Comment:
How about replacing this part with `if all(isinstance(s, int) for s in slices)`? It's more idiomatic and readable. Your code is slightly faster, but the difference is negligible.
The same goes for PR #2420.